### PR TITLE
added gyp project files

### PR DESCRIPTION
**Pull request by dstulic**

### CODE REVIEW

File: gyp/configure.sh [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        "
+
+# android generator does not support --generator-output
+GYP_DEFINES=${DEFINES} ANDROID_BUILD_TOP=SRCDIR gyp lua.gyp --check --depth . -f android-make 
+
+# fix the path scope of the android make file
+/bin/cat > $SRCDIR/output/lua/jni/Android.mk << "END_OF_LOCAL_PATH_FIX"
+# do not edit, generated by gyp
+LUA_PROJECT_PATH:= $(realpath $(join $(dir $(call this-makefile)), /../../../.. ))
+#LUA_PROJECT_PATH:= $(LUA_PROJECT_PATH)/lua-5.2.2/src
```
Comment (fivemicro) [inline gyp/configure.sh]: do you need these two commented out lines?

File: gyp/lua.gyp [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+          ],
+        }
+      },
+    ],
+    [ 
+      "OS=='ios'", 
+      {
+        'xcode_settings': {
+          'SDKROOT': 'iphoneos',
+          #'TARGETED_DEVICE_FAMILY': '1,2',
```
Comment (fivemicro) [inline gyp/lua.gyp]: Do you need these commented out lines?

File: gyp/make.sh [hunk 3]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+
+SRCDIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+cd $SRCDIR
+
+# mac
+#$SRCDIR/output/mac/lua.xcodeproj
```
Comment (fivemicro) [inline gyp/make.sh]: Do you need these commented out lines?

File: gyp/output/lua/jni/Application.mk [hunk 4]
```diff
@@ -0,0 +1,38 @@
+# select target arch
+#APP_ABI := armeabi armeabi-v7a x86 mips
```
Comment (fivemicro) [inline gyp/output/lua/jni/Application.mk]: Do you need these commented out lines?

File: gyp/output/lua/jni/Application.mk [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+NDK_TOOLCHAIN_VERSION := clang3.5
+# GCC versions available
+#NDK_TOOLCHAIN_VERSION := 4.6
+#NDK_TOOLCHAIN_VERSION := 4.8
+
+
+# target android platform
+# min version for pthread_rwlock_t
+APP_PLATFORM := android-9
+#APP_PLATFORM := android-18
```
Comment (fivemicro) [inline gyp/output/lua/jni/Application.mk]: Why not android-18 or later?

File: gyp/output/lua/jni/Application.mk [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+# enable c++11 extentions in source code
+# using c++_std will enable this by default
+# clang and gcc 4.8
+#APP_CPPFLAGS += -std=c++11
+# gcc 4.6
+#APP_CPPFLAGS += -std=c++0x
+
+# enable c11
+APP_CFLAGS += -std=c11
```
Comment (fivemicro) [inline gyp/output/lua/jni/Application.mk]: In the .gyp file above I see this line in compiler_c_flags.  Do you need to repeat it here?

File: gyp/output/lua/jni/Application.mk (hunk 5 above)
Comment (dstulic) [inline gyp/output/lua/jni/Application.mk]: android 18 is what the andrive is build for, lua's min framework for correct compile is 9. so no need to raise it here. the andrive project can set it up to higher level to override this setting.

File: gyp/output/lua/jni/Application.mk (hunk 6 above)
Comment (dstulic) [inline gyp/output/lua/jni/Application.mk]: you are right, probably not, i will think about what the right way to do this is.. additionally they can collide. and not really sure which setting would win..

**Comment by dstulic** [comment]
hey @fivemicro this is proof of concept. i need those comments for a bit longer until we are ready to rol this out

**Comment by dstulic** [comment]
@fivemicro also not ready to merge, just need feedback on the gyp setup

**Pull request merged**